CART/CARTLib/examples/MultiContrastSegmentationEvaluationDataUnit.py

```diff
@@ -0,0 +1,160 @@
+        # Validate inputs then load resources
+        self._validate()
+        self._initialize_resources()
+
+    def to_dict(self) -> dict[str, str]:
+        """Serialize back to case_data format."""
+        output = {key: self.case_data[key] for key in self.volume_keys}
+        output[self.SEGMENTATION_KEY] = self.case_data[self.SEGMENTATION_KEY]
+        output[self.COMPLETED_KEY] = self.is_complete
+        return output
+
+    def focus_gained(self) -> None:
+        """Show all volumes and segmentation when this unit gains focus."""
+        for node in self.volume_nodes.values():
+            node.SetDisplayVisibility(True)
+        self.segmentation_node.SetDisplayVisibility(True)
+        self._set_subject_shown(True)
+
+    def focus_lost(self) -> None:
+        """Hide all volumes and segmentation when focus is lost."""
+        print(f"{self.validated=}")
+        for node in self.volume_nodes.values():
+            node.SetDisplayVisibility(False)
+        self.segmentation_node.SetDisplayVisibility(False)
+        self._set_subject_shown(False)
+
+    def clean(self) -> None:
+        """Clean up the hierarchy node and its children."""
+        super().clean()
+        if self.subject_id is not None:
+            self.hierarchy_node.RemoveItem(self.subject_id)
+
+    def _validate(self) -> None:
+        """
+        Ensure that all discovered volume keys and the segmentation key
+        refer to existing files.
+        """
+        for key in self.volume_keys:
+            self.validate_key_is_file(key)
+        self.validate_key_is_file(self.SEGMENTATION_KEY)
+        self.validated = True
+
+    def validate_key_is_file(self, key: str) -> None:
+        """
+        Confirm that case_data[key] exists, is a path under data_path,
+        and refers to a file.
+        """
+        rel_path = self.case_data.get(key)
+        if not rel_path:
+            raise ValueError(f"Case {self.uid} missing required entry '{key}'.")
+        full_path = self.data_path / rel_path
+        if not full_path.exists():
+            raise ValueError(f"Path for '{key}' does not exist: {full_path}")
+        if not full_path.is_file():
+            raise ValueError(f"Path for '{key}' is not a file: {full_path}")
+
+    def _initialize_resources(self) -> None:
+        """
+        Load all volumes and the segmentation into MRML nodes,
+        sync geometry, and create a subject hierarchy.
+        """
+        if not getattr(self, "validated", False):
+            raise ValueError(_("Data must be validated before initializing resources."))
+
+        primary_node = self._init_volume_nodes()
+        seg_node = self._init_segmentation_node()
+
+        # Align segmentation to primary volume geometry
+        seg_node.SetReferenceImageGeometryParameterFromVolumeNode(primary_node)
+
+        # Group in subject hierarchy
+        self.subject_id = create_subject(self.uid, primary_node, seg_node)
```

Comment:
This should fix (or at least reduce) the memory leak.
```suggestion
        # Group in subject hierarchy
        self.subject_id = create_subject(self.uid, self.segmentation_node, *self.volume_nodes)
```